src/scales/quantitative.js

```diff
@@ -168,9 +170,16 @@ export function ScaleThreshold(key, channels, {
   range = interpolate !== undefined ? quantize(interpolate, domain.length + 1) : registry.get(key) === color ? ordinalRange(scheme, domain.length + 1) : undefined,
   reverse
 }) {
+  const descending = domain[0] > domain[1];
```

Comment:
We’d need to use `order(domain)` here again. But as I hinted at in the other comment, I don’t think we need to add support for descending domains for the threshold scales; we only need to do it for quantize scales. Threshold scales require you to specify explicit thresholds rather than an extent, so it seems reasonable to require that those thresholds are in ascending order.

Also, it looks like ScaleThreshold doesn’t correctly handle when the _domain_ is specified as an iterable rather than an array (because it references _domain_.length), and it probably should.